### PR TITLE
Documentation - Improvising with examples

### PR DESCRIPTION
**Pull request by 0xtejas**
I have planned to do it in the following manner:

- [x]  Bring the vol3 tutorial similar to vol2.
- [x]  Have examples for each plugin, which wasn't provided in vol3 wiki.

As of now the Linux page has been updated similar to vol2 wiki. Soon more updates will come

This PR is wrt issue #562

### CODE REVIEW

**Comment by 0xtejas** [comment]
Please don't close the PR until I let you know guys, this PR was made in order to get reviews and do the needful fixes as I continue to work on. 

Thanks

**Comment by ikelos** [comment]
No problem, thanks for working on this!  We'll leave this open, and keep an eye on it to see when it's done...  5:)

File: doc/source/Linux.rst
Comment (ikelos) [inline doc/source/Linux.rst]: Probably better not to recommend a specific tool.  Happy to explain how an example tool might work (such as lime), but best not to play favourites if possible.

File: doc/source/Linux.rst
Comment (ikelos) [inline doc/source/Linux.rst]: Ideally we don't want to have to update this each time a new linux plugin is added, so either finish with with `...` so people know that there's potentially more and/or make it explicit in the description that this is just an example list and not necessarily a complete one.

File: doc/source/Linux.rst
Comment (ikelos) [inline doc/source/Linux.rst]: I'd kind of prefer to just show users how they'd find this information from the tool, rather than including a list (that will get out of date) in the documentation.

File: doc/source/Linux.rst
Comment (ikelos) [inline doc/source/Linux.rst]: There's a few articles missing here and there.  This would sound more natural as `To create a symbol table please`.  I'll probably do a full grammar scan of the documentation once it's all done, so don't worry about it too much at the moment.

**Review by ikelos — COMMENTED** [review]
Looking good, sorry for doing a review so quickly, it just feels like a way of making sure the task isn't too big when all the rest of the documentation is added.  5:)

**Comment by digitalisx** [comment]
Thank you for your work @tejas15802,
Awesome, I think documentation is one of the really good contributions! 😃

I don't fully knowing the PR policy of volatility 3 not yet,
If continuous changes are needed for the goal, why don't you use Draft PR?

Of course, Draft PR can proceed with the review, and if it is changed to Open,
it means that you are ready for the final review. 🙌

**Comment by 0xtejas** [comment]
@ikelos I ll resolve them in further commits.

**Comment by 0xtejas** [comment]
> Thank you for your work @tejas15802, Awesome, I think documentation is one of the really good contributions! 😃
> 
> I don't fully knowing the PR policy of volatility 3 not yet, If continuous changes are needed for the goal, why don't you use Draft PR?
> 
> Of course, Draft PR can proceed with the review, and if it is changed to Open, it means that you are ready for the final review. 🙌

Sorry I wasn't aware of it. I'll change it to draft

**Comment by digitalisx** [comment]
> > Thank you for your work @tejas15802, Awesome, I think documentation is one of the really good contributions! 😃
> > I don't fully knowing the PR policy of volatility 3 not yet, If continuous changes are needed for the goal, why don't you use Draft PR?
> > Of course, Draft PR can proceed with the review, and if it is changed to Open, it means that you are ready for the final review. 🙌
> 
> Sorry I wasn't aware of it. I change it to draft

Oh, it wasn't your fault or mistake, it was just a suggestion. 😅 Thank you for reflecting my opinion!

File: doc/source/Linux.rst
Comment (digitalisx) [inline doc/source/Linux.rst]: I think it's a familiar name for many people to write "LiME".

**Comment by tejas15802** [comment]
Do you guys want me to have something like this [one](https://github.com/volatilityfoundation/volatility/wiki/Command-Reference)

**Comment by 0xtejas** [comment]
Please review the PR and let me know if there any improvisations needed
@ikelos @digitalisx 

Thanks

File: doc/source/Linux.rst
Comment (digitalisx) [inline doc/source/Linux.rst]: Shouldn't the path name used in the document be unified like `/home/users`, `just like README.md`?

File: doc/source/Linux.rst
Comment (digitalisx) [inline doc/source/Linux.rst]: Isn't it necessary to distinguish it as `volatility 3`?

File: doc/source/Linux.rst
Comment (digitalisx) [inline doc/source/Linux.rst]: I know this ISF service is very useful. :)
However, rather than connecting to an external project that is not officially managed in the official document, I think it is better to create your own symbol table and provide a way to resolve the issue.

File: doc/source/Linux.rst
Comment (digitalisx) [inline doc/source/Linux.rst]: I think this would be good for unification.
```suggestion
    $ python3 vol.py --help | grep -i linux. | head -n 5
```


And commands over the pipeline are also very useful. However, I think it would be better to tell you the basic commands in the document as much as possible and omit them with the "..." mark. It is inefficient to show all results, but at least the user can see them through the command.

File: doc/source/Linux.rst
Comment (digitalisx) [inline doc/source/Linux.rst]: I think it's similar to the context mentioned above. I know it can be a good example of education! :)
But taking a dataset that's not officially managed and using it means that when the link expires or is not valid, we have to renew it. Therefore, it seems better to leave the data set for the example to the users.

File: doc/source/Windows.rst
Comment (digitalisx) [inline doc/source/Windows.rst]: Same intention as above. (external project) :)

File: doc/source/Windows.rst
Comment (digitalisx) [inline doc/source/Windows.rst]: Same context as above. (external resource) :)

File: doc/source/Windows.rst
Comment (digitalisx) [inline doc/source/Windows.rst:22]: Same context as above. (Pipeline commands) :)

File: doc/source/Windows.rst
Comment (digitalisx) [inline doc/source/Windows.rst]: It seems that the version of Volatility is different for each document. When explaining the examples, I think it would be better to unify the versions. Also, if it is a document that will be explained based on many people, it seems better to use the last release, version `2.0.1`.

File: doc/source/Windows.rst
Comment (digitalisx) [inline doc/source/Windows.rst]: Same context as above. (version) :)

File: doc/source/Windows.rst
Comment (digitalisx) [inline doc/source/Windows.rst:101]: Same context as above. (version) :)

File: doc/source/symbol-tables.rst
Comment (digitalisx) [inline doc/source/symbol-tables.rst:41]: Is there any particular reason to change this?

**Review by digitalisx — COMMENTED** [review]
Thank you for your efforts @tejas15802,
I am happy that the work is finished and ready for review.

As you requested, I reviewed the contents of the work once and left an opinion on it. Since I am not the maintainer of this project, my view on the document may not be justified.

Please check it out and see if there are any improvements that can help. 🙌

File: doc/source/Linux.rst
Comment (ikelos) [inline doc/source/Linux.rst]: `demonstration on suite` seems like unusual english, what are you trying to express here?  `some demonstration of several of the plugins available from`?

Also, this sentence just stops without saying where from...  5;)

File: doc/source/Linux.rst
Comment (ikelos) [inline doc/source/Linux.rst]: This may get old if we eventually add arm/other architectures.  Not a bad thing, more a comment to myself to keep it in mind.

File: doc/source/Linux.rst
Comment (ikelos) [inline doc/source/Linux.rst]: Might be better to use relative paths, since this doesn't seem to be under the just extracted directory (`lime-forensics-1.1-r14` isn't mentioned anywhere)

File: doc/source/Linux.rst
Comment (ikelos) [inline doc/source/Linux.rst]: This could do with more of a description, particularly if we're recommending people to run commands as root.

File: doc/source/Linux.rst
Comment (ikelos) [reply doc/source/Linux.rst]: I agree, you do provide attribution, but it should be as a separate footnote or note, rather than something directly recommended in the main documentation.

File: doc/source/Linux.rst
Comment (ikelos) [inline doc/source/Linux.rst]: Listing plugins should be avoided, it goes out of date quickly.  At best this is a list of *some* of the linux plugins.  Saying htere will be more available suggests this was a complete list and dates the documentation, better to just say it's s sample of the available plugins please.

File: doc/source/Linux.rst
Comment (ikelos) [inline doc/source/Linux.rst:28]: There's a fair number of things like missing articles (please create an issue with *a* description of the plugin).  Rather than pointing them all out, I'd suggest that after merge, I do a correction pass is done to ensure the grammar is all accurate.

File: doc/source/Linux.rst
Comment (ikelos) [reply doc/source/Linux.rst]: Yes please!  This is the expected method of invocation, we can't guarantee the package unpacks as an executable.

File: doc/source/Linux.rst
Comment (ikelos) [inline doc/source/Linux.rst]: There are other `<global options>` like `-f`, probably best to include mention of them as you do `plugin_option`.  Also, as with `<path to memory image>`, I'd suggest surrounding it with angle brackets to show it's not supposed to be typed verbatim (so `<plugin name> <plugin option>`.

File: doc/source/Linux.rst
Comment (ikelos) [inline doc/source/Linux.rst:79]: As pointed out above, this may change.  There's currently work going on to change the requirement of where the ISF files (see #630).

File: doc/source/conf.py
Comment (ikelos) [inline doc/source/conf.py:114]: How does this affect the rest of the documentation?  A little description of why this is here (just in the PR comments) would be useful.

**Review by ikelos — COMMENTED** [review]
Hiya, a few minor style changes, but mostly pretty good.  It will need a grammar pass once it gets committed but otherwise looks good, thanks!  5:)

File: doc/source/conf.py
Comment (0xtejas) [reply doc/source/conf.py:114]: That helped me to tag titles from other pages, if I remember correctly. It was long back I added that. But am sure it was for cross referencing.

File: doc/source/Linux.rst
Comment (0xtejas) [reply doc/source/Linux.rst]: Note regarding the sudo has been added

File: doc/source/symbol-tables.rst
Comment (0xtejas) [reply doc/source/symbol-tables.rst:41]: > Is there any particular reason to change this?

Yes it's because I was not able to cross reference the heading when it had slash.

File: doc/source/Windows.rst
Comment (0xtejas) [reply doc/source/Windows.rst:101]: All versions are changed to 2.0.1 as suggested

File: doc/source/Windows.rst
Comment (0xtejas) [reply doc/source/Windows.rst]: Description on why the pipe is used added in note section

File: doc/source/Linux.rst
Comment (0xtejas) [reply doc/source/Linux.rst]: Removed the links

File: doc/source/Linux.rst
Comment (0xtejas) [reply doc/source/Linux.rst]: Made some changes, hope its correct

**Comment by 0xtejas** [comment]
Just pinging to notify, if the last notification missed out.
@ikelos

**Comment by digitalisx** [comment]
Hello @tejas15802,

If you haven't responded to Github's review request, it's a good idea to wait at least a month. (unless it's very urgent)

Perhaps your work completion and review request has already been recognized by him, If time permits he will be happy to look at it if it is available or reasonable to prioritize. It's his [feedback](https://github.com/volatilityfoundation/volatility3/pull/663#issuecomment-1058494921) on a similar request.

We all have to wait a little bit more broadly for a healthy open-source ecosystem. 🙂

**Comment by 0xtejas** [comment]
okay noted, will wait then. 
I apologize ikelos.

File: doc/source/Linux.rst
Comment (ikelos) [inline doc/source/Linux.rst]: available from...?

File: doc/source/Linux.rst
Comment (ikelos) [inline doc/source/Linux.rst]: Will these instructions get out of date?  I'm wary about including specific implementation instructions which may get old/stale, given how often documentation gets reviewed/refreshed.

File: doc/source/Windows.rst
Comment (ikelos) [inline doc/source/Windows.rst]: Probably good to offer a few possibilities to avoid the suggestion of favouritism.

File: doc/source/Windows.rst
Comment (ikelos) [inline doc/source/Windows.rst:50]: This isn't quite accurate.  We always require an ISF file, but often volatility will determine which one we need and generate it from debugging information available from microsoft automatically.

**Review by ikelos — COMMENTED** [review]
A few major bits to fix up (sentences that don't end) and one sentence that was inaccurate, but once those get tweaked I'll push it and then fix up the grammar...  5:)

File: doc/source/Linux.rst
Comment (0xtejas) [reply doc/source/Linux.rst]: I'm not sure if LIME will be working on future architecture. So Should I remove the line specifying the architecture?

File: doc/source/Windows.rst
Comment (0xtejas) [reply doc/source/Windows.rst:50]: I meant it works most of the time that we don't have to create a ISF file manually like Linux

File: doc/source/Linux.rst
Comment (ikelos) [reply doc/source/Linux.rst]: It's not really whether they work with a future architecture, it's more whether they decide to move to cmake or meson or some other build system that makes these instructions incorrect.  I know it's not as "one stop" as providing direct instructions for compilation in our documentation, but I'd much prefer directing people at the packages documentation for figuring out how to compile it, than including the information ourselves.  Either that, or state that these commands are representative of compilation, but you should always read the package's documentation to ensure you're using the latest instructions.

File: doc/source/Linux.rst
Comment (ikelos) [inline doc/source/Linux.rst]: Again, as with windows, it might be worth mentioning AVML as an alternative.

**Review by ikelos — COMMENTED** [review]
Thanks for the update, a couple more bits and then I'll probably push and tidy some grammatical things...

File: doc/source/Linux.rst
Comment (0xtejas) [reply doc/source/Linux.rst]: ~~AVML is for Linux - Acquire Volatile Memory for Linux~~
edit: later realized what you meant. Will do the relevant updates

Comment (ikelos) [reply doc/source/Linux.rst]: It is, you're right, this is under the linux section and in comparison to LiME the LInux Memory Extractor.

Comment (ikelos) [reply doc/source/Linux.rst]: I meant in comparison to windows where we're going to mention both winpmem and also another imager (such as FTK imager or similar).

**Comment by 0xtejas** [comment]
How about now?

**Review by ikelos — APPROVED** [review]
Yep, I'll merge it and then make some grammatical fixes.  Thanks  5:)

**Comment by 0xtejas** [comment]
Thank you : )

**Pull request merged**
